packages/react-native-executorch/common/rnexecutorch/models/text_to_image/TextToImage.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include <set>
+#include <utility>
+
+#include <executorch/extension/tensor/tensor_ptr.h>
+#include <jsi/jsi.h>
+#include <opencv2/opencv.hpp>
+
+#include <ReactCommon/CallInvoker.h>
+
+#include <rnexecutorch/jsi/OwningArrayBuffer.h>
+#include <rnexecutorch/metaprogramming/ConstructorHelpers.h>
+#include <rnexecutorch/models/BaseModel.h>
+#include <rnexecutorch/models/embeddings/text/TextEmbeddings.h>
+
+#include <rnexecutorch/models/text_to_image/Constants.h>
+#include <rnexecutorch/models/text_to_image/Decoder.h>
+#include <rnexecutorch/models/text_to_image/Scheduler.h>
+#include <rnexecutorch/models/text_to_image/UNet.h>
+
+namespace rnexecutorch {
+namespace models::text_to_image {
+using namespace facebook;
+
+using executorch::aten::Tensor;
+using executorch::extension::TensorPtr;
+
+class TextToImage {
+public:
+  TextToImage(const std::string &tokenizerSource, float schedulerBetaStart,
+              float schedulerBetaEnd, int32_t schedulerNumTrainTimesteps,
+              int32_t schedulerStepsOffset, const std::string &encoderSource,
+              const std::string &unetSource, const std::string &decoderSource,
+              int32_t imageSize,
+              std::shared_ptr<react::CallInvoker> callInvoker);
+  std::shared_ptr<OwningArrayBuffer> generate(std::string input,
+                                              size_t numInferenceSteps);
+  size_t getMemoryLowerBound() const noexcept;
+  void unload() noexcept;
+
+private:
+  std::shared_ptr<OwningArrayBuffer>
+  postprocess(const std::vector<float> &output);
```

Comment:
And in implementation ;)
```suggestion
  postprocess(const std::vector<float> &output) const;
```